lib/engines/hcktest/tools.rb

```diff
@@ -387,9 +393,11 @@ def apply_project_filters(project)
       end
     end
 
-    def zip_test_result_logs(test_id, target_key, machine, tag)
+    def zip_test_result_logs(test_id, target_key, machine, tag, index: 1, index_instance_id: false) # rubocop:disable Metrics/ParameterLists
```

Comment:
I guess you meant:
```suggestion
    def zip_test_result_logs(test_id, target_key, machine, tag, index: -1, index_instance_id: false) # rubocop:disable Metrics/ParameterLists
```

But I wonder if it is a good idea to have default arguments or even to name parameters here; it's possible to use three dots to make this simpler and remove the `Metrics/ParameterLists` warning:

```suggestion
    def zip_test_result_logs(...)
```

The default arguments or parameter names can be provided in rtoolsHCK instead.

Reply:
>even to name parameters here

yes, because Tools#zip_test_result_logs and RToolsHCK#zip_test_result_logs have partially different parameters. 

t.zip_test_result_logs(index, test_id, target_key, **tag**, machine, **tag**, index_instance_id:) - tag 2 times

Reply:
`index` fixed

Reply:
I wonder why we fiddle with parameters here instead of dealing with them at rtoolsHCK or `AutoHCK::Tests`. I particularly have an intuition that moving the logic to `AutoHCK::Tests` will result in simpler code.